nanopub/client.py

```diff
@@ -227,3 +230,48 @@ def claim(self, text, rdftriple=None):
 
         nanopub = Publication.from_assertion(assertion_rdf=assertion_rdf)
         self.publish(nanopub)
+
+    def _check_public_keys_match(self, uri):
+        """ Check for matching public keys of a nanopublication with the profile.
+
+        Raises:
+            AssertionError: When the nanopublication is signed with a public key that does not
+                match the public key in the profile
+        """
+        publication = self.fetch(uri)
+        their_public_keys = list(publication.pubinfo.objects(rdflib.URIRef(uri + '#sig'),
+                                                             namespaces.NPX.hasPublicKey))
+        if len(their_public_keys) > 0:
```

Comment:
Can this list in principle be more than length 1? Should we check and warn if that is the case?

Reply:
Yes, it's rdf so in principle anything can be the case...

Reply:
I added a warning, but still taking the first in the list.